src/Common/MemoryBufferWriter.cs

```diff
@@ -307,5 +313,27 @@ protected override void Dispose(bool disposing)
                 Reset();
             }
         }
+
+        /// <summary>
+        /// Holds a byte[] from the pool and a size value. Basically a Memory but guaranteed to be backed by an ArrayPool byte[], so that we know we can return it.
+        /// </summary>
+        private struct BufferSegment
+        {
+            public byte[] Buffer { get; }
```

Comment:
Why are these public? Seems like you'd only ever need to access the Span or Memory

Reply:
There is one place where it saves LOC by making these public. In `CopyToSlowAsync(Stream)` we don't have to do the crazy `#ifdef`s to use the Stream Span APIs. We know this is always an array though so I guess I can try unwrapping it.

Reply:
Except that requires exposing a `ReadOnlyMemory<T>` and exposing both a `ROM` and a `ROS` just to avoid making fields on a private nested struct public seems unncessary.